launch_testing_ros/launch_testing_ros/wait_for_topics.py

```diff
@@ -120,13 +136,16 @@ def start_subscribers(self, topic_tuples):
                     10
                 )
             )
+            # Initialize ring buffer of received messages
+            self.received_messages_buffer[topic_name] = deque(maxlen=self.max_number_of_messages)
 
     def callback_template(self, topic_name):
 
         def topic_callback(data):
             if topic_name not in self.received_topics:
                 self.get_logger().debug('Message received for ' + topic_name)
                 self.received_topics.add(topic_name)
```

Comment:
Ah, right I missed that. The logic here means that this will trigger only once. @LastStarDust you'll have to change this section, and it would be nice to check for the buffer length in the test case later.

Reply:
I believe the logic should be : Keep waiting until there is at least one message on each of the required topics. If any of the topics have at least one msg, push them in a deque.

Reply:
@adityapande-1995 Sorry but I am not following. Isn't that what is already happening?
Moreover, that logic should not be affected by anything I have changed in the source code.

```python
            if topic_name not in self.received_topics:
                self.get_logger().debug('Message received for ' + topic_name)
                self.received_topics.add(topic_name)  # the received topic is added to the list of received topics
                self.received_messages_buffer[topic_name].append(data) # at the same time the received message 
                                                                       # is added to the list of received messages
            if self.received_topics == self.expected_topics: # if all the expected topics are received, interrupt the 
                                                             # wait and return.
                self.msg_event_object.set()
```

Reply:
You are right. I will fix it as soon as possible.

Reply:
Fixed in commit 7c9e181fb3c4a3257f5315bfe217a4748019f0c3